packages/browser-wallet/src/background/index.ts

```diff
@@ -4,11 +4,44 @@ import {
     InternalMessageType,
     MessageType,
 } from '@concordium/browser-wallet-message-hub';
-import { storedSelectedAccount, storedUrlWhitelist } from '@shared/storage/access';
+import { storedSelectedAccount, storedUrlWhitelist, storedJsonRpcUrl } from '@shared/storage/access';
 
+import { v4 as uuidv4 } from 'uuid';
 import bgMessageHandler from './message-handler';
 import { forwardToPopup, HandleMessage, HandleResponse, RunCondition, setPopupSize } from './window-management';
 
+// TODO Replace with concordiumSDK HttpProvider when cross-fetch supports service workers. https://github.com/lquixada/cross-fetch/issues/78
```

Comment:
Could the HttpProvider be refactored to work in a service worker?

Reply:
https://github.com/Concordium/concordium-node-sdk-js/pull/82